src/uri.jl

```diff
@@ -153,46 +153,24 @@ function Base.isvalid(uri::URI)
 end
 
 # RFC3986 Unreserved Characters (and '~' Unsafe per RFC1738).
-@inline shouldencode(c) = !((c >= UInt8('A') && c <= UInt8('Z'))
-                         || (c >= UInt8('a') && c <= UInt8('z'))
-                         || (c >= UInt8('0') && c <= UInt8('9'))
-                         || c == UInt8('-')
-                         || c == UInt8('.')
-                         || c == UInt8('_'))
-hexstring(x) = string('%', uppercase(hex(x,2)))
+@inline isunsafe(c::Char) = !(c in "-._") && (!isascii(c) || !isalnum(c))
```

Comment:
Do you have a performance test for this stuff?
If not, perhaps if you could provide a use-case example that captures your performance concerns I could add a performance test and ensure that I'm not causing significant regressions.